### ISSUE TITLE
Placeholders and colors not populating as expected.

### ISSUE THREAD

**Issue by loudambiance** [OGBlackDiamond/Proxy-Messages#1]
In trying to setup Proxy-Messages, I've run into an issue where colors don't work and the {cur} tag is not populated on join and leave. It also appears to always select the first message.

My config looks like:
```yaml
join-message-options:
  - "{#00ff00}>>> {cur} >>>:{#ffffff} {player} joined"
  - "{#00ff00}>>> {cur} >>>:{#ffffff} OMG EVERYONE! {player} graced us with their pressence!"
  - "{#00ff00}>>> {cur} >>>:{#ffffff} {player} is here, now the party can start!"
  - "{#00ff00}>>> {cur} >>>:{#ffffff} SHHHHH, {player} is here, everybody hide"

leave-message-options:
  - "{#ff0000}<<< {cur} <<<:{#ffffff} {player} left"
  - "{#ff0000}<<< {cur} <<<:{#ffffff} Bye Felicia! I mean {player}."
  - "{#ff0000}<<< {cur} <<<:{#ffffff} Parties over, {player} left."

switch-message-options:
  - "{#0000FF}{prev} >>> {cur}:{#ffffff} {player} changed their mind"
```

and causes output like:
![image](https://github.com/user-attachments/assets/ebb44a73-159b-4f84-85ff-d0f25ce0cb43)
and
![image](https://github.com/user-attachments/assets/fb322928-9f36-4546-957f-1f0d697443f9)

**Comment by loudambiance** [comment]
2.2.0 fixed everything except the ability to use {cur} outside of the switch scope

**Comment by OGBlackDiamond** [comment]
glad 2.2.0 fixed it, thats the first thing I was going to ask. I do not currently support `{cur}` outside of switch scope. I can work on that though, I'll close this issue when I get to it. 

Also, you may want to save your current configuration, then delete it and let a new one regenerate. I added some configuration stuff for Discord support in that update and ProxyMessages might behave in unexpected ways if it doesn't see those config options :)

**Comment by loudambiance** [comment]
I noticed that. Thanks for that! I would ask if you could make the discord embeds have configurable colors based on the type of event, join, leave, switch. When I disabled the DiscordSRV events and enabled the Proxy-Messages ones, I went from having green for join, and red for leave, to just blue for all events:

![image](https://github.com/user-attachments/assets/8de6815f-e7a8-4c2f-bef6-28953b50486a)
to
![image](https://github.com/user-attachments/assets/77580719-b9d1-440f-922e-28c825105fa2)

If you don't mind, I may fork your repo and start work that and the support for {cur} and send it back as a pull request.

**Comment by OGBlackDiamond** [comment]
Yeah I wouldn't mind at all, I can also work on it today if you don't mind waiting. This is still very actively in development and I'm trying to add stuff all the time haha. 

The colors should be fairly easy I'll add that too.

**Comment by loudambiance** [comment]
> Yeah I wouldn't mind at all, I can also work on it today if you don't mind waiting. This is still very actively in development and I'm trying to add stuff all the time haha.
> 
> The colors should be fairly easy I'll add that too.
> 
> Also if you don't mind me asking - where did you find my plugin?

I found it on Modrinth. I nearly have the color changes done.

**Comment by OGBlackDiamond** [comment]
okie dokie. 

Thanks, how did you find it, were you searching for anything in particular?

Sorry if these are strange questions I'm just fairly new to plugin development so I'm just curious.

**Comment by loudambiance** [comment]
I was browsing by recently published, saw the description and it matched a problem I needed to solve after switching from Bungee to Velocity.

**Comment by OGBlackDiamond** [comment]
very cool, thanks

**Comment by loudambiance** [comment]
Added a pull request for the discord embed colors

**Comment by OGBlackDiamond** [comment]
thanks
resolved by #2
